src/as-cache.c

```diff
@@ -1995,3 +1995,16 @@ as_cache_search (AsCache *cache, const gchar * const *terms, gboolean sort, GErr
 
 	return g_steal_pointer (&results);
 }
+
+gboolean
+as_cache_get_is_empty(AsCache* cache)
```

Comment:
Please make this return a component count instead:
```suggestion
guint
as_cache_get_component_count (AsCache* cache)
```